app/src/main/java/nu/yona/app/utils/AppUtils.java

```diff
@@ -288,25 +289,37 @@ public static String getHourDigit(String digit)
 	 */
 	public static void reportException(String className, Exception e, Thread t, DataLoadListener listener)
 	{
-		ErrorMessage errorMessage = new ErrorMessage();
-		if (e != null && e.getMessage() != null)
+		ErrorMessage errorMessage = getErrorMessage(e);
+		if (listener != null)
 		{
-			errorMessage.setMessage(e.getMessage());
+			listener.onError(errorMessage);
 		}
 		else
 		{
-			errorMessage.setMessage(YonaApplication.getAppContext().getString(R.string.error_message));
+			showErrorToast(errorMessage);
+			logExceptionToCrashlytics(e);
 		}
-		if (listener != null)
+	}
+
+	/**
+	 * getErrorMessage from give Exception or return a generic error message
+	 */
+
+	public static ErrorMessage getErrorMessage(Exception e)
+	{
+		ErrorMessage errorMessage = new ErrorMessage();
+		if (e != null && e.getMessage() != null)
 		{
-			listener.onError(errorMessage);
+			errorMessage.setMessage(e.getMessage());
+			Logger.loge(e.getClass().getSimpleName(), e.getMessage());
 		}
 		else
 		{
-			showErrorToast(errorMessage);
-			logExceptionToCrashlytics(e);
+			String message = YonaApplication.getAppContext().getString(R.string.generic_error_message);
+			errorMessage.setMessage(message);
+			Logger.loge(e.getClass().getSimpleName(), message);
 		}
-		Logger.loge(e.getClass().getSimpleName(), e.getMessage());
+		return errorMessage;
```

Comment:
Please make the method this simple:
```
	public static ErrorMessage getErrorMessage(Exception e)
	{
		ErrorMessage errorMessage = new ErrorMessage();
		if (e != null && e.getMessage() != null)
		{
			return new ErrorMessage(e.getMessage());
		}
		return new ErrorMessage(YonaApplication.getAppContext().getString(R.string.generic_error_message));
	}
```

Please also add ``ErrorMessage`` to your refactoring list: this is a perfect class for immutable behavior. Just remove all setters and only allow taking the values into the constructor.